src/data/finsupp/lattice.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson
+-/
+import data.finsupp
+import tactic.squeeze
+import algebra.ordered_group
+--open finsupp
+
+/-!
+# Lattice structure on finsupps
+
+This file provides instances of ordered structures on finsupps.
+
+-/
+
+open_locale classical
+noncomputable theory
+variables {α : Type*} {β : Type*} [has_zero β] {μ : Type*} [canonically_ordered_add_monoid μ]
+variables {γ : Type*} [canonically_linear_ordered_add_monoid γ]
+
+namespace finsupp
+
+lemma le_def [partial_order β] {a b : α →₀ β} : a ≤ b ↔ ∀ (s : α), a s ≤ b s := by refl
+
+instance : order_bot (α →₀ μ) :=
+{ bot := 0, bot_le := by simp [finsupp.le_def, ← bot_eq_zero], .. finsupp.partial_order}
+
+/-- Used to construct binary operations on `finsupp`s -/
```

Comment:
Include the `to_fun` field in the doc-string. (doc-strings appear in pop-up tooltips, so it's important they're readable in isolation).